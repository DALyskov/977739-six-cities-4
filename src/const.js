export const APPROVED_NAME = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];
