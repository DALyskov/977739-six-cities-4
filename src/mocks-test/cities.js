export const cities = [`Amsterdam`, `Paris`];
