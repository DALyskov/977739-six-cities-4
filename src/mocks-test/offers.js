export const offers = [
  {
    bedrooms: 1,
    city: {
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
      name: `Amsterdam`,
    },
    description: `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
    features: [
      `Coffee machine`,
      `Laptop friendly workspace`,
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Breakfast`,
      `Dishwasher`,
      `Towels`,
      `Washer`,
      `Washing machine`,
    ],

    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    hostId: 25,

    id: 5,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    isBookmark: false,
    isPremium: false,

    location: {
      latitude: 52.397540000000006,
      longitude: 4.9099759999999995,
      zoom: 16,
    },

    maxAdults: 2,
    previewImg: `img/room.jpg`,
    price: 101,
    rating: 3.5,
    name: `Perfectly located Castro`,
    type: `room`,
  },

  {
    bedrooms: 5,
    city: {
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
      name: `Amsterdam`,
    },
    description: `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
    features: [
      `Coffee machine`,
      `Laptop friendly workspace`,
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Breakfast`,
      `Dishwasher`,
      `Towels`,
      `Washer`,
      `Washing machine`,
    ],

    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    hostId: 25,

    id: 1,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    isBookmark: false,
    isPremium: false,

    location: {
      latitude: 52.397540000000006,
      longitude: 4.886976,
      zoom: 16,
    },

    maxAdults: 6,
    previewImg: `img/room.jpg`,
    price: 813,
    rating: 2.4,
    name: `Wood and stone place`,
    type: `house`,
  },

  {
    bedrooms: 4,
    city: {
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
      name: `Amsterdam`,
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    features: [`Breakfast`, `Laptop friendly workspace`],

    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    hostId: 25,

    id: 14,
    images: [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
    isBookmark: false,
    isPremium: true,

    location: {
      latitude: 52.397540000000006,
      longitude: 4.886976,
      zoom: 16,
    },

    maxAdults: 7,
    previewImg: `img/room.jpg`,
    price: 423,
    rating: 4.7,
    name: `Penthouse, 4-5 rooms + 5 balconies`,
    type: `hotel`,
  },
];

export const originOffers = [
  {
    bedrooms: 1,
    city: {
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
      name: `Amsterdam`,
    },
    description: `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
    goods: [
      `Coffee machine`,
      `Laptop friendly workspace`,
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Breakfast`,
      `Dishwasher`,
      `Towels`,
      `Washer`,
      `Washing machine`,
    ],

    host: {
      'avatar_url': 'img/avatar-angelina.jpg',
      id: 25,
      'is_pro': true,
      name: 'Angelina',
    },
    id: 5,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    'is_favorite': false,
    'is_premium': false,
    location: {
      latitude: 52.397540000000006,
      longitude: 4.9099759999999995,
      zoom: 16,
    },
    'max_adults': 2,
    'preview_image': `img/room.jpg`,
    price: 101,
    rating: 3.5,
    title: `Perfectly located Castro`,
    type: `room`,
  },
];
