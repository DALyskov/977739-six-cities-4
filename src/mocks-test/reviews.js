export const reviews = [
  {
    comment:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    date: `2020-07-08T16:06:01.819Z`,
    id: 1,
    rating: 4,
    userAvatar: 'img/avatar-max.jpg',
    userId: 4,
    isUserPro: false,
    userName: `Max`,
  },
  {
    comment:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    date: `2020-07-08T16:06:01.819Z`,
    id: 2,
    rating: 3.5,
    userAvatar: `img/avatar-max.jpg`,
    userId: 3,
    isUserPro: true,
    userName: `Lex`,
  },
];

export const originReviews = [
  {
    comment:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    date: `2020-07-08T16:06:01.819Z`,
    id: 1,
    rating: 4,
    user: {
      'avatar_url': 'img/avatar-max.jpg',
      id: 4,
      'is_pro': false,
      name: `Max`,
    },
  },
  {
    comment:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    date: `2020-07-08T16:06:01.819Z`,
    id: 2,
    rating: 3.5,
    user: {
      'avatar_url': 'img/avatar-max.jpg',
      id: 3,
      'is_pro': true,
      name: `Lex`,
    },
  },
];
