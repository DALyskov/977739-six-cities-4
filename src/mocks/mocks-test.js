export const offers = [
  {
    id: 1,
    isPremium: true,
    images: [
      `img/apartment-01.jpg`,
      `img/room.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    price: 120,
    isBookmark: false,
    rating: 4.5,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    bedrooms: 3,
    maxAdults: 1,
    features: [
      'Heating',
      'Kitchen',
      'Cable TV',
      'Washing machine',
      'Coffee machine',
    ],
    hostName: `Dima`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: true,
    description:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: 'Amsterdam',
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8,
    },
  },
  {
    id: 2,
    isPremium: false,
    images: [`img/room.jpg`],
    price: 80,
    isBookmark: true,
    rating: 4,
    name: `Wood and stone place`,
    type: `Private room`,
    bedrooms: 2,
    maxAdults: 3,
    features: [
      'Heating',
      'Kitchen',
      'Cable TV',
      'Washing machine',
      'Coffee machine',
      'Dishwasher',
    ],
    hostName: `Alex`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: false,
    description:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: 'Amsterdam',
    },
    location: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 8,
    },
  },
  {
    id: 3,
    isPremium: false,
    images: [`img/apartment-02.jpg`],
    price: 132,
    isBookmark: false,
    rating: 4,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    bedrooms: 5,
    maxAdults: 2,
    features: ['Heating', 'Cable TV', 'Coffee machine', 'Dishwasher'],
    hostName: `Pavel`,
    hostAvatar: `img/avatar-angelina.jpg`,
    isHostPro: false,
    description:
      'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: 'Amsterdam',
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198,
      zoom: 8,
    },
  },
];

export const incompletePlace = {
  id: 1,
  images: [
    `img/apartment-01.jpg`,
    `img/room.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
  ],
  price: 120,
  rating: 4.5,
  name: `Beautiful & luxurious apartment at great location`,
  type: `Apartment`,
  bedrooms: 3,
  maxAdults: 1,
  features: [
    'Heating',
    'Kitchen',
    'Cable TV',
    'Washing machine',
    'Coffee machine',
  ],
  hostName: `Dima`,
  hostAvatar: `img/avatar-angelina.jpg`,
  description:
    'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: 'Amsterdam',
  },
  location: {
    latitude: 52.3909553943508,
    longitude: 4.85309666406198,
    zoom: 8,
  },
};
