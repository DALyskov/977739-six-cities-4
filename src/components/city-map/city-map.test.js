import React from 'react';
import renderer from 'react-test-renderer';

import {offers} from '../../mocks/mocks-test.js';

import {MapClassName} from '../../const.js';
import CityMap from './city-map.jsx';

describe(`CityMap_snapchots`, () => {
  it(`with_data`, () => {
    const tree = renderer
      .create(<CityMap offersByCity={offers} className={MapClassName.MAIN} />, {
        createNodeMock: () => {
          return document.createElement(`div`);
        },
      })
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
