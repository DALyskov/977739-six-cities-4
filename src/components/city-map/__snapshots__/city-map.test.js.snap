// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityMap_snapchots with_data 1`] = `
<section
  className="cities__map map"
/>
`;
